### PR TITLE
[SPARK-30881][SQL][DOCS]Revise the doc of spark.sql.sources.parallelPartitionDiscovery.threshold

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Revise the doc of SQL configuration `spark.sql.sources.parallelPartitionDiscovery.threshold`.
### Why are the changes needed?

The doc of configuration "spark.sql.sources.parallelPartitionDiscovery.threshold" is not accurate on the part "This applies to Parquet, ORC, CSV, JSON and LibSVM data sources".

We should revise it as effective on all the file-based data sources.

### Does this PR introduce any user-facing change?

No

### How was this patch tested?

None. It's just doc.

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 1]
```diff
@@ -878,8 +878,8 @@ object SQLConf {
     buildConf("spark.sql.sources.parallelPartitionDiscovery.threshold")
       .doc("The maximum number of paths allowed for listing files at driver side. If the number " +
         "of detected paths exceeds this value during partition discovery, it tries to list the " +
-        "files with another Spark distributed job. This applies to Parquet, ORC, CSV, JSON and " +
-        "LibSVM data sources.")
+        "files with another Spark distributed job. This configuration is effective only when " +
+        "using file-based sources such as Parquet, JSON and ORC.")
```
Comment (gengliangwang) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: This follows the doc of `spark.sql.files.maxPartitionBytes` and `spark.sql.files.openCostInBytes`

**Comment by gengliangwang** [comment]
retest this please.

**Comment by maropu** [comment]
retest this please

**Comment by gengliangwang** [comment]
@maropu I think the Mima test failure is from https://github.com/apache/spark/pull/27395

**Comment by maropu** [comment]
Ah, I see...

**Comment by gengliangwang** [comment]
retest this please.

**Review by cloud-fan — APPROVED** [review]

**Review by maropu — APPROVED** [review]

**Comment by cloud-fan** [comment]
retest this please

**Comment by gengliangwang** [comment]
Thanks for the review.
Merging to master/3.0

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala (hunk 1 above)
Comment (stevegomez17) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: spark.sql.sources.parallelPartitionDiscovery.threshold.

**Pull request closed without merging**